### ISSUE TITLE
Customizable syntax-highlighting themes

### ISSUE THREAD

**Issue by fzzr-** [uranusjr/macdown#273]
Please if possible implement this feature. Customizable syntax highlighting themes need for user can add missing languages like "Haxe". Github have highlighting theme for Haxe :)

**Comment by uranusjr** [comment]
Actually you already can—Just add your language definition inside `Resources/Prism/components`, and edit `Resources/Prism/components.js` to add an entry for it.

But as you might find, it is not exactly trivial to write a language definition. I don’t really want to let users configure those rules at will because most either can’t or don’t want to. And if you are among those who  can and are willing to, you should probably contribute to [Prism](https://github.com/LeaVerou/prism) directly. Once your language rule is merged, I will update MacDown asap to include it. :)

Closing as wontfix

**Comment by fzzr-** [comment]
Okay. Maybe need to copy Prism/components to Application Support and scan changes? Is it possible?

**Comment by fzzr-** [comment]
It because if I change/add custom syntax in the application bundle then after next app-update I'll lose all my changes.

∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙
Please do not change the subject when replying.
Sincerely, Alexander Kozlovskij
Alexander@Kozlovskij.ru
+7 (915) 363 8813
+1 (858) 997-0013

**Comment by fzzr-** [comment]
Or maybe add list into the pref-window for adding custom syntaxes. See how similar feature implemented for scripts and custom actions in TextWrangler for example.

**Comment by uranusjr** [comment]
Copying Prism to Application Support is more complex than other things because we depend on an external repository, and can’t control anticipate how it’s updated because the update happens.

I’m currently working on a more robust resource-loading schema (#229), and when that’s finished it should be possible to load things from Application Support without needing to copy the main Prism stuff. My current plan is to finish most of it during the Chinese New Year holidays (later this month), but I can’t promise anything yet.

Also, even after that is done, I would still encourage everyone to contribute syntax definitions to Prism. They will welcome them. :)
